### PR TITLE
v6 build pipeline update

### PR DESCRIPTION
**Pull request by ribeiroguilherme**
## Summary
- Upgraded to Rollup 4
- Upgraded other dev dependencies as well
- Set Node to run on version 18.18

**Fixed issue**:

### CODE REVIEW

File: .github/workflows/.size-limit.yml [hunk 1]
```diff
@@ -8,14 +8,15 @@ jobs:
 
     steps:
       - name: Checkout Repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```
Comment (sponglord) [inline .github/workflows/.size-limit.yml:11]: Out of interest what does this `v4` refer to?

**Review by sponglord — APPROVED** [review]

File: packages/lib/.stylelintrc.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
         "no-descending-specificity": null,
         "selector-class-pattern": null,
         "scss/no-global-function-names": null,
-        "declaration-block-no-redundant-longhand-properties": null
+        "declaration-block-no-redundant-longhand-properties": null,
+        "media-feature-range-notation": null
```
Comment (sponglord) [inline packages/lib/.stylelintrc.json:8]: What does this new rule do?

File: .github/workflows/.size-limit.yml (hunk 1 above)
Comment (ribeiroguilherme) [reply .github/workflows/.size-limit.yml:11]: Version 4 of the Github actions/checkout : https://github.com/actions/checkout

**Review by longyulongyu — APPROVED** [review]

File: packages/lib/.stylelintrc.json (hunk 2 above)
Comment (ribeiroguilherme) [reply packages/lib/.stylelintrc.json:8]: https://stylelint.io/user-guide/rules/media-feature-range-notation/

As I updated stylelint, it started complaining about some media rules. I disabled the rule as it didn't seem like a big deal. Let me know if you think the opposite

**Pull request merged**